src/ddsrt/src/expand_vars.c

```diff
@@ -95,7 +95,9 @@ static char *expand_varbrace (const char **src, expand_fn expand, expand_lookup_
         goto err;
     }
     name = ddsrt_malloc ((size_t) (*src - start) + 1);
-    memcpy (name, start, (size_t) (*src - start));
```

Comment:
Here, `src > start` necessarily holds under the assumption that the input is a null-terminated string starting with `{`, that this is the case is easily proven (short of arbitrary memory corruption). That makes the conditional always true and so it shouldn't be present. (And besides, there's another MISRA rule (rule 14.3) that forbids such a conditional!)

I don't know (yet) how to suppress the warning from codeql, but that's what we need to do here.

Reply:
codeql doesn't detect violations here. but another tool detects it as an violation!🤷